File: client/sdk/python/openai/responses.py
"""OpenAI /v1/responses endpoint functions."""

from openai import OpenAI


def create_client(api_key, base_url):
    return OpenAI(api_key=api_key, base_url=base_url + "/v1")


def create_response(prompt, model, api_key, base_url):
    client = create_client(api_key=api_key, base_url=base_url)
    response = client.responses.create(
        model=model,
        input=prompt,
    )
    output_text = getattr(response, "output_text", None)
    if output_text:
        return output_text

    texts = []
    for item in getattr(response, "output", []) or []:
        for content in getattr(item, "content", []) or []:
            text = getattr(content, "text", None)
            if text:
                texts.append(text)
    return "".join(texts)


def stream_response(prompt, model, api_key, base_url):
    client = create_client(api_key=api_key, base_url=base_url)
    stream = client.responses.create(
        model=model,
        input=prompt,
        stream=True,
    )
    for event in stream:
        if getattr(event, "type", "") == "response.output_text.delta":
            delta = getattr(event, "delta", None)
            if delta:
                yield delta

File: client/sdk/typescript/src/openai/responses.ts
import type { ClientConfig } from "../config";

import { createClient } from "./shared";

export async function createResponse(
  config: ClientConfig,
  prompt: string,
  model: string,
): Promise<string> {
  const client = createClient(config);
  const response = await client.responses.create({
    model,
    input: prompt,
  });

  if (typeof response.output_text === "string" && response.output_text) {
    return response.output_text;
  }

  const texts: string[] = [];
  for (const item of response.output ?? []) {
    if (!("content" in item) || !Array.isArray(item.content)) {
      continue;
    }
    for (const content of item.content) {
      if ("text" in content && typeof content.text === "string" && content.text) {
        texts.push(content.text);
      }
    }
  }
  return texts.join("");
}

export async function* streamResponse(
  config: ClientConfig,
  prompt: string,
  model: string,
): AsyncIterable<string> {
  const client = createClient(config);
  const stream = await client.responses.create({
    model,
    input: prompt,
    stream: true,
  });

  for await (const event of stream) {
    if (event.type === "response.output_text.delta" && typeof event.delta === "string" && event.delta) {
      yield event.delta;
    }
  }
}
